homeassistant/components/sensibo/climate.py

```diff
@@ -187,129 +209,29 @@ def _do_update(self, data):
-
-    @property
-    def target_temperature_step(self):
-        """Return the supported step of target temperature."""
-        if self.temperature_unit == self.hass.config.units.temperature_unit:
-            # We are working in same units as the a/c unit. Use whole degrees
-            # like the API supports.
-            return 1
-        # Unit conversion is going on. No point to stick to specific steps.
-        return None
-
-    @property
-    def hvac_mode(self):
-        """Return current operation ie. heat, cool, idle."""
-        if not self._ac_states["on"]:
-            return HVAC_MODE_OFF
-        return SENSIBO_TO_HA.get(self._ac_states["mode"])
-
-    @property
-    def current_humidity(self):
-        """Return the current humidity."""
-        return self._measurements["humidity"]
-
-    @property
-    def current_battery(self):
-        """Return the current battery voltage."""
-        return self._measurements.get("batteryVoltage")
-
-    @property
-    def current_temperature(self):
-        """Return the current temperature."""
-        # This field is not affected by temperatureUnit.
-        # It is always in C
-        return convert_temperature(
-            self._measurements["temperature"], TEMP_CELSIUS, self.temperature_unit
-        )
-
-    @property
-    def hvac_modes(self):
-        """List of available operation modes."""
-        return self._operations
-
-    @property
-    def fan_mode(self):
-        """Return the fan setting."""
-        return self._ac_states.get("fanLevel")
-
-    @property
-    def fan_modes(self):
-        """List of available fan modes."""
-        return self._current_capabilities.get("fanLevels")
-
-    @property
-    def swing_mode(self):
-        """Return the fan setting."""
-        return self._ac_states.get("swing")
-
-    @property
-    def swing_modes(self):
-        """List of available swing modes."""
-        return self._current_capabilities.get("swing")
-
-    @property
-    def name(self):
-        """Return the name of the entity."""
-        return self._name
-
-    @property
-    def min_temp(self):
-        """Return the minimum temperature."""
-        return (
+        self._attr_min_temp = (
             self._temperatures_list[0] if self._temperatures_list else super().min_temp
         )
-
-    @property
-    def max_temp(self):
-        """Return the maximum temperature."""
-        return (
+        self._attr_max_temp = (
             self._temperatures_list[-1] if self._temperatures_list else super().max_temp
         )
+        self._attr_temperature_unit = self._temperature_unit
 
-    @property
-    def unique_id(self):
-        """Return unique ID based on Sensibo ID."""
-        return self._id
+        self._attr_supported_features = 0
+        for key in self._ac_states:
+            if key in FIELD_TO_FLAG:
+                self._attr_supported_features |= FIELD_TO_FLAG[key]
+
+        self._attr_state = self._external_state or super().state
```

Comment:
This is not allowed. The climate platform should only implement the `hvac_mode` attribute. Not `state`.

Reply:
This one is a bit non-standard since it existing for `async_assume_state`.  It would be better to convert this to use DataUpdateCoordinator and use `async_set_updated_data` here instead.